core/opendaq/signal/include/opendaq/signal_impl.h

```diff
@@ -670,4 +677,84 @@ ErrCode SignalBase<TInterface, Interfaces...>::setStreamed(Bool streamed)
     return OPENDAQ_IGNORED;
 }
 
+template <typename TInterface, typename... Interfaces>
+ErrCode SignalBase<TInterface, Interfaces...>::getValue(IBaseObject ** value)
+{
+    OPENDAQ_PARAM_NOT_NULL(value);
+    std::scoped_lock lock(this->sync);
+
+    if (lastDataPacket.assigned() && lastDataPacket.getSampleCount())
+    {
+        const auto & descriptor = lastDataPacket.getDataDescriptor();
+
+        if (descriptor.getDimensions().getCount() != 0)
+            return OPENDAQ_IGNORED;
+
+        auto idx = lastDataPacket.getSampleCount() - 1;
+
+        switch (descriptor.getSampleType())
+        {
+            case SampleType::Float32:
+            {
+                auto data = static_cast<float*>(lastDataPacket.getData());
```

Comment:
I'm wondering if `getData` should also apply the resolution. Your thoughts @dejancrn ?